src/nrnpython/nrnpy_p2h.cpp

```diff
@@ -355,8 +360,11 @@ static void hpoasgn(Object* o, int type) {
         "HOC cannot handle PythonObject assignment with more than one index.");
   }
   Py_DECREF(poright);
-  //	hoc_push_object(o);
   hoc_stkobj_unref(o);
+  if (err) {
+    PyErr_Print();
+    hoc_execerror("Assignment to PythonObject failed", NULL);
```

Comment:
This always prints a Python error. However, hoc_execerrors are presently not suppressable
in terms of printing in this context and the Python message is needed to give the hoc error
some diagnostic value. In this context hoc is calling some pythc c-api functions to do its work
and the python interpreter is not being used, per se.  This can be revisited when the
printing of hoc_execerrors become suppressible (or saved for possible later printing) when the hoc_execerror occurs within a python try block. I'd have to figure out how that could be known from c-api calls. There must be one.